src/rgw/rgw_json_enc.cc

```diff
@@ -1494,3 +1494,76 @@ void RGWOrphanSearchState::dump(Formatter *f) const
   encode_json("stage", stage, f);
   f->close_section();
 }
+
+void RGWObjTags::dump(Formatter *f) const
+{
+  for (const auto& tag: tag_map){
+    f->open_object_section("tag_map");
+    f->dump_string("Key", tag.first);
+    f->dump_string("Value", tag.second);
+    f->close_section();
+  }
+}
+
+void lc_op::dump(Formatter *f) const
+{
+  encode_json("status", status, f);
+  encode_json("dm_expiration", dm_expiration, f);
+  encode_json("expiration", expiration, f);
+  encode_json("noncur_expiration", noncur_expiration, f);
+  encode_json("mp_expiration", mp_expiration, f);
+  if (expiration_date != boost::none) {
+    utime_t ut(*expiration_date);
+    encode_json("expiration_date", ut, f);
+  }
+  if (obj_tags != boost::none) {
+    encode_json("obj_tags", *obj_tags, f);
+  }
+}
+
+void LCFilter::dump(Formatter *f) const
+{
+  encode_json("prefix", prefix, f);
+  encode_json("obj_tags", obj_tags, f);
+}
+
+void LCExpiration::dump(Formatter *f) const
+{
+  encode_json("days", days, f);
+  encode_json("date", date, f);
+}
+
+void LCRule::dump(Formatter *f) const
+{
+  encode_json("id", id, f);
+  encode_json("prefix", prefix, f);
+  encode_json("status", status, f);
+  encode_json("expiration", expiration, f);
+  encode_json("noncur_expiration", noncur_expiration, f);
+  encode_json("mp_expiration", mp_expiration, f);
+  encode_json("filter", filter, f);
+  encode_json("dm_expiration", dm_expiration, f);
+}
+
+void RGWLifecycleConfiguration::dump(Formatter *f) const
+{
+  map<string, lc_op>::const_iterator iter = prefix_map.begin();
+  f->open_array_section("prefix_map");
+  for (; iter != prefix_map.end(); ++iter) {
+    f->dump_string("prefix", iter->first);
```

Comment:
the items in the array shoudl all have the same type.  if you're trying to make a map, do open_object_section outside the loop, and inside the loop to open_object_section(iter->first.c_str()).

Reply:
Thanks. I have correct it. 
But I have got the same mismatch type  in both RGWObjManifest::dump  and RGWAccessControlList::dump;Should I pull another new request to correct that ?

Reply:
sorry @liewegas  , It's my mistake. I misunderstood you suggestion,  I have correct it.